pallets/tendermint-light-client/src/types.rs

```diff
@@ -0,0 +1,753 @@
+    /// Validators for the next block
+    pub next_validators_hash: Hash,
+    /// Consensus params for the current block
+    pub consensus_hash: Hash,
+    /// State after txs from the previous block
+    /// AppHash is usually a SHA256 hash, but in reality it can be any kind of data
+    #[cfg_attr(
+        feature = "std",
+        serde(deserialize_with = "deserialize_string_as_bytes")
+    )]
+    pub app_hash: Vec<u8>,
+    /// Root hash of all results from the txs from the previous block
+    pub last_results_hash: Option<Hash>,
+    /// Hash of evidence included in the block
+    pub evidence_hash: Option<Hash>,
+    /// Original proposer of the block
+    pub proposer_address: TendermintAccountId,
+}
+
+impl HeaderStorage {
+    pub fn new(
+        version: VersionStorage,
+        chain_id: Vec<u8>,
+        height: u64,
+        timestamp: TimestampStorage,
+        last_block_id: Option<BlockIdStorage>,
+        last_commit_hash: Option<Hash>,
+        data_hash: Option<Hash>,
+        validators_hash: Hash,
+        next_validators_hash: Hash,
+        consensus_hash: Hash,
+        app_hash: Vec<u8>,
+        last_results_hash: Option<Hash>,
+        evidence_hash: Option<Hash>,
+        proposer_address: TendermintAccountId,
+    ) -> Self {
+        Self {
+            version,
+            chain_id,
+            height,
+            timestamp,
+            last_block_id,
+            last_commit_hash,
+            data_hash,
+            validators_hash,
+            next_validators_hash,
+            consensus_hash,
+            app_hash,
+            last_results_hash,
+            evidence_hash,
+            proposer_address,
+        }
+    }
+}
+
+/// Represents  UTC time since Unix epoch
+#[derive(Encode, Decode, Clone, RuntimeDebug, TypeInfo, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct TimestampStorage {
+    pub seconds: i64,
+    pub nanos: u32,
+}
+
+impl TimestampStorage {
+    pub fn new(seconds: i64, nanos: u32) -> Self {
+        Self { seconds, nanos }
+    }
+}
+
+impl TryFrom<TimestampStorage> for Time {
+    type Error = tendermint::Error;
+
+    fn try_from(value: TimestampStorage) -> Result<Self, Self::Error> {
+        Time::from_unix_timestamp(value.seconds, value.nanos)
+    }
+}
+
+/// CommitSig represents a signature of a validator.
+/// It's a part of the Commit and can be used to reconstruct the vote set given the validator set.
+#[derive(Encode, Decode, Clone, RuntimeDebug, TypeInfo, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize))]
+pub enum CommitSignatureStorage {
+    /// no vote was received from a validator.
+    BlockIdFlagAbsent,
+    /// voted for the Commit.BlockID.
+    BlockIdFlagCommit {
+        /// Validator address
+        validator_address: TendermintAccountId,
+        /// Timestamp of vote
+        timestamp: TimestampStorage,
+        /// Signature of vote
+        signature: TendermintVoteSignature,
```

Comment:
I've noticed that while working on the mappings the other way around - in Tendermint this field is optional. It should be optiona here as well.

Reply:
done